src/java/org/apache/cassandra/net/MessagingService.java

```diff
@@ -1664,4 +1676,113 @@ public static boolean isEncryptedConnection(InetAddressAndPort address)
         }
         return true;
     }
+
+    public void blockForPeers()
+    {
+        // TODO make these yaml props?
+        int alivePercent = Integer.getInteger(Config.PROPERTY_PREFIX + "blockForPeers.percent", 70);
+        if (alivePercent < 0)
```

Comment:
This is fair, but I felt that giving *some* upper bound didn't seem unreasonable. Should we really delay starting the process by an extra order of magnitude just because an operator added an extra zero to the configuration? I understand your point about least surprise, but that might be beyond the spirit of this ticket. wdyt?

Reply:
Well I think it really sucks when you think you are getting the benefit from some config option and your not and you don't know or don't know why. Out of scope arguments aren't really applicable because this is the ticket adding the feature driven by these config options.

I am not sure what you mean by order of magnitude. The zero clamp means someone tried to enable this feature, but didn't and so their stuff is silently unreliable.

The 100 clamp is they meant to set a value < 100 but had an extra digit and didn't and it's silently being slower to start. I doubt they are going to care as much about that in production use cases.

I don't think you'll find many operators that complain about being informed via a clear error message their config is not sane. This is also an option that people have to request explicitly in order to get an invalid value to clamp in the first place so I really do think they want to know.

Reply:
Just to clarify: you do not like the bounding of `aliveTimeoutSecs`, correct? `alivePercent` is just bounding the percentage.

Reply:
I don't like silently clamping any of them. If you feel strongly about upper bounding aliveTimeoutSecs it's the closest to being OK because the DB still comes up and is reliable at up to 100 seconds and most of the time it won't wait that long since it will get to 70%. If someone sets a value higher than that we "know" it's not useful so clamping it doesn't hurt them.

Reply:
My issue isn't with having a bound it's with enforcing it silently. Some of the time you can pick a value for them, but not all of the time.

Reply:
hmm, starting thinking about this, and, yeah, I guess we can leave the upper bound unenforced and instead just log that the operator can choose to ignore